Modify changelog generator functionality (#1585)
kyma-project/kyma: tools/changelog-generator/README.md

1. Create a `CHANGELOG.md` file under the specified absolute path to the repository. Use this command:

```bash

> docker run --rm -v {absolutePathToRepository}:/repository -w /repository -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFilePath} -e CONFIG_FILE={configFilePath} -e SKIP_REMOVING_LATEST=true changelog-generator sh /app/generate-full-changelog.sh --configure-git

Reviewer: Please rename `LATEST_VERSION` to something like `NEW_RELEASE_TITLE` or something like that - as it is only used for title in changelog, right?

Author: It is used to changelog file tittle, in commit message (during pull changelog file to root repository) and if "FROM_TAG" is used then learna-changelog got it as a "to" argument. I changed the name.

Reviewer: > if "FROM_TAG" is used then learna-changelog got it as a "to" argument

Hmm, but to be honest it should be used in this way, as by design the tag shouldn't exist yet 😛

```

Replace values in curly braces with proper details, where:
- `{absolutePathToRepository}` is the absolute path to the repository.
- `{applicationVersion}` is the currently released application version.
- `{githubToken}` is the GitHub API token with the read-only access to the repository.
- `{sshFilePath}` is the path to the SSH file used for Git to authenticate with the repository.
- `{configFilePath}` is the path to file used by `lerna-changelog` to repository configuration

> Add `SKIP_REMOVING_LATEST=true` variable if you want skip removing 'latest' tag functionality.

Reviewer: Please describe configuration for `SKIP_REMOVING_LATEST like in above paragraph.

So in the example above it should be something like 
`-e SKIP_REMOVING_LATEST={skipRemovingLatest}` and then below you describe the role of the variable

```

Replace values in curly braces with proper details, where:
- `{absolutePathToRepository}` is the absolute path to the repository.
- `{currentBranch}` is the current Git branch name.
- `{releaseTitle}` is the currently released application title uses as title in file or in commit message.
- `{sshFilePath}` is the path to the SSH file used for Git to authenticate with the repository.

### Generate the latest release changelog

To generate a changelog for a single release that contains merged pull requests for the latest Kyma version, run this command:

```bash
docker run --rm -v {absolutePathToRepository}:/repository -w /repository -e FROM_TAG={previousTag} -e TO_TAG={latestTag} -e NEW_RELEASE_TITLE={releaseTitle} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} -e CONFIG_FILE={configFilePath} -e SKIP_REMOVING_LATEST={skipRemovingLatest} changelog-generator sh /app/generate-release-changelog.sh --configure-git
```

Replace values in curly braces with proper details, where:
- `{absolutePathToRepository}` is the absolute path to the repository.

> - `{previousTag}` optionally, one tag before last. If provided changelog will be generated from `{previousTag}` to `{latestTag}`. If not provided changelog will be generated from the one tag before latest used tag.

Reviewer 2: ```suggestion
- `{previousTag}` specifies the first tag which is included in the changelog. If not provided, the changelog is generated from the tag which precedes the latest tag used.
```

> - `{latestTag}` optionally, is tag to which changelog is created. If provided changelog will be generated to `{latestTag}`. If not provided changelog will be generated to latest used tag.

Reviewer 2: ```suggestion
- `{latestTag}` specifies the last tag which is included in the changelog. If not provided, the changelog is generated to the latest tag used.
```

To generate a changelog for a single release that contains merged pull requests for the latest Kyma version, run this command:

```bash
docker run --rm -v /path/to/repository/:/repository -w /repository -e FROM_TAG={previousTag} -e LATEST_VERSION={applicationVersion} -e GITHUB_AUTH={githubToken} -e SSH_FILE={sshFile} -e CONFIG_FILE={configFilePath} changelog-generator sh /app/generate-release-changelog.sh --configure-git
```

Replace values in curly braces with proper details, where:
- `{absolutePathToRepository}` is the absolute path to the repository.

> - `{previousTag}` optionally, one tag before last. If provided changelog will be generated from `{previousTag}` to `{applicationVersion}`

Reviewer: It should be mentioned, that if not provided, the latest tag is used.

- `{applicationVersion}` is the currently released application version.
- `{githubToken}` is the GitHub API token with the read-only access to the repository.
- `{sshFilePath}` is the path to the SSH file used for Git to authenticate with the repository.
- `{configFilePath}` is the path to file used by `lerna-changelog` to repository configuration

> Add `SKIP_REMOVING_LATEST=true` variable if you want skip removing 'latest' tag functionality.

Reviewer: Please describe configuration for `SKIP_REMOVING_LATEST like in above paragraph.

So in the example above it should be something like 
`-e SKIP_REMOVING_LATEST={skipRemovingLatest}` and then below you describe the role of the variable
